test/app_automate_test.rb

```diff
@@ -0,0 +1,279 @@
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, false)
+    appium_mock.expect(:click, true)
+    appium_mock.expect(:find_element, appium_mock) do |locator, element_id|
+      locator === :accessibility_id && element_id === "test_button"
+    end
+
+    driver.click_element("test_button")
+
+    appium_mock.verify
+
+  ensure
+    driver.instance_variable_set(:@driver, nil)
+  end
+
+  def test_background_app_nil
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    appium_mock = Minitest::Mock.new
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, true)
+
+    driver.background_app
+
+    appium_mock.verify
+
+  ensure
+    driver.instance_variable_set(:@driver, nil)
+  end
+
+  def test_background_app_defaults
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    appium_mock = Minitest::Mock.new
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, false)
+    appium_mock.expect(:background_app, true) do |timeout|
+      timeout === 3
+    end
+
+    driver.background_app
+
+    appium_mock.verify
+
+  ensure
+    driver.instance_variable_set(:@driver, nil)
+  end
+
+  def test_background_app_custom_timeout
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    appium_mock = Minitest::Mock.new
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, false)
+    appium_mock.expect(:background_app, true) do |timeout|
+      timeout === 5
+    end
+
+    driver.background_app(5)
+
+    appium_mock.verify
+
+  ensure
+    driver.instance_variable_set(:@driver, nil)
+  end
+
+  def test_reset_app_nil
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    appium_mock = Minitest::Mock.new
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, true)
+
+    driver.reset_app
+
+    appium_mock.verify
+
+  ensure
+    driver.instance_variable_set(:@driver, nil)
+  end
+
+  def test_reset_app
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    appium_mock = Minitest::Mock.new
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    appium_mock.expect(:nil?, false)
+    appium_mock.expect(:reset, true)
+
+    driver.reset_app
+
+    appium_mock.verify
+
+  ensure
+    driver.instance_variable_set(:@driver, nil)
+  end
+
+  def test_upload_app_success
+    driver = AppAutomateDriver.new(USERNAME, ACCESS_KEY, LOCAL_ID)
+    appium_mock = Minitest::Mock.new
+    appium_mock.expect(:nil?, false)
+    driver.instance_variable_set(:@driver, appium_mock)
+
+    json_response = JSON.dump({
+      :app_url => "App_url"
+    })
+
+    command_call = Minitest::Mock.new
+    command_call.expect(:call, json_response) do |cmd|
+      cmd === "curl -u \"#{USERNAME}:#{ACCESS_KEY}\" -X POST \"https://api-cloud.browserstack.com/app-automate/upload\" -F \"file=@#{APP_LOCATION}\""
```

Comment:
Use `%()` so you don't need to escape the double quotes (https://github.com/rubocop-hq/ruby-style-guide#percent-q-shorthand)